ScreenCapturerExample/ExampleWebViewSource.swift

```diff
@@ -93,38 +93,99 @@ class ExampleWebViewSource: NSObject {
-         * Note that this technique is inefficient because it causes an extra malloc / copy to occur for every frame.
-         * For a more performant solution, provide a pool of buffers and use them to back a CGBitmapContext.
+         * The underlying data is marked as immutable, but we are the sole owner and can do as we please ...
+         * Also, the CVPixelBuffer constructor will only accept a mutable pointer.
          */
-        let image: CGImage? = contextImage?.cgImage
-        let dataProvider: CGDataProvider? = image?.dataProvider
-        let data: CFData? = dataProvider?.data
-        let baseAddress = CFDataGetBytePtr(data!)
-        contextImage = nil
+        let mutableBaseAddress = UnsafeMutablePointer<UInt8>(mutating: baseAddress)
+        let pixelFormat: TVIPixelFormat
+
+        switch byteOrderInfo {
+        case .byteOrder32Little:
+            // Encountered on iOS simulators.
+            // Note: We do not account for the pre-multiplied alpha leaving the images too dim.
+            // This problem could be solved using vImageUnpremultiplyData_RGBA8888 to operate in-place on the pixels.
+            assert(alphaInfo == .premultipliedFirst || alphaInfo == .noneSkipFirst)
+            pixelFormat = TVIPixelFormat.format32BGRA
+        case .byteOrder32Big:
+            // Never encountered with snapshots on iOS, but maybe on macOS?
+            assert(alphaInfo == .premultipliedFirst || alphaInfo == .noneSkipFirst)
+            pixelFormat = TVIPixelFormat.format32ARGB
+        case .byteOrder16Little:
+            pixelFormat = TVIPixelFormat.format32BGRA
+            assert(false)
+        case .byteOrder16Big:
+            pixelFormat = TVIPixelFormat.format32BGRA
+            assert(false)
+        default:
+            // The pixels are formatted in the default order for CoreGraphics, which on iOS is kCVPixelFormatType_32RGBA.
+            // This format is included in Core Video for completeness, and creating a buffer returns kCVReturnInvalidPixelFormat.
+            // We will instead repack the memory from RGBA to BGRA, which is supported by Core Video (and Twilio Video).
+            // Note: While UIImages captured on a device claim to have pre-multiplied alpha, the alpha channel is always opaque (0xFF).
+            pixelFormat = TVIPixelFormat.format32BGRA
+            assert(alphaInfo == .premultipliedLast || alphaInfo == .noneSkipLast)
+
+            for row in 0 ..< cgImage.height {
+                let rowByteAddress = mutableBaseAddress.advanced(by: row * cgImage.bytesPerRow)
+
+                for pixel in stride(from: 0, to: cgImage.width * 4, by: 4) {
+                    // Swap the red and blue channels.
+                    let red = rowByteAddress[pixel]
+                    rowByteAddress[pixel] = rowByteAddress[pixel + 2]
+                    rowByteAddress[pixel+2] = red
+                }
+            }
```

Comment:
Consider using vImagePermuteChannels in order to swap red/blue.

https://stackoverflow.com/questions/10654909/converting-bgra-to-argb

Reply:
Done, and this is much faster.